daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/preprocessing/ValueTranslator.scala

```diff
@@ -243,10 +337,36 @@ private[lf] final class ValueTranslator(
   }
 
   // This does not try to pull missing packages, return an error instead.
-  def translateValue(
+  // TODO: https://github.com/digital-asset/daml/issues/17082
+  //  This is used by script and trigger, this should problaby use ValueTranslator.Config.Legacy
+  def strictTranslateValue(
```

Comment:
`strict` is not a great prefix here, since `Config.Legacy` has `allowFieldReordering = true`,
so is in some ways less strict. (I know you mean _strict_ as in _doesn't allow upgrade_)

Perhaps we should call it `legacyTranslateValue`.

Reply:
It is really the case that the legacy usage in `daml-script` etc requires the flexibility of field reordering?
Assuming so, do we have any plans to address this?

Reply:
> It is really the case that the legacy usage in daml-script etc requires the flexibility of field reordering?

This actually why I called it "strict" I assume daml-script/trigger does note require the flexibility. 

> Assuming so, do we have any plans to address this?

As soon we have a bit of time.